core/src/test/java/org/opensearch/sql/expression/datetime/TimeStampAddTest.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.expression.datetime;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.sql.data.model.ExprDateValue;
+import org.opensearch.sql.data.model.ExprDatetimeValue;
+import org.opensearch.sql.data.model.ExprIntegerValue;
+import org.opensearch.sql.data.model.ExprNullValue;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTimeValue;
+import org.opensearch.sql.data.model.ExprTimestampValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.DSL;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.ExpressionTestBase;
+import org.opensearch.sql.expression.FunctionExpression;
+
+class TimeStampAddTest extends ExpressionTestBase {
+
+  private static Stream<Arguments> getTestDataForTimestampAdd() {
+    return Stream.of(
+        Arguments.of("MINUTE", 1, new ExprStringValue("2003-01-02 00:00:00"),
+            "2003-01-02 00:01:00"),
+        Arguments.of("WEEK", 1, new ExprStringValue("2003-01-02 00:00:00"),
+            "2003-01-09 00:00:00"),
+        //Date
+        Arguments.of("MINUTE", 1, new ExprDateValue("2003-01-02"),
+            "2003-01-02 00:01:00"),
+        Arguments.of("WEEK", 1, new ExprDateValue("2003-01-02"),
+            "2003-01-09 00:00:00"),
+        //Datetime
+        Arguments.of("MINUTE", 1, new ExprDatetimeValue("2003-01-02 00:00:00"),
+            "2003-01-02 00:01:00"),
+        Arguments.of("WEEK", 1, new ExprDatetimeValue("2003-01-02 00:00:00"),
+            "2003-01-09 00:00:00"),
+        //timstamp
```

Comment:
Fix spelling. Add white space between groups of tests.
```suggestion
        //Timestamp
```